src/Analyzers/CSharp/CodeFixes/ConvertNamespace/ConvertNamespaceTransform.cs

```diff
@@ -243,28 +337,59 @@ private static FileScopedNamespaceDeclarationSyntax ConvertNamespaceDeclaration(
             return fileScopedNamespace;
         }
 
-        private static NamespaceDeclarationSyntax ConvertFileScopedNamespace(FileScopedNamespaceDeclarationSyntax fileScopedNamespace)
+        private static NamespaceDeclarationSyntax ConvertFileScopedNamespace(ParsedDocument document, FileScopedNamespaceDeclarationSyntax fileScopedNamespace, string lineEnding, NewLinePlacement newLinePlacement)
         {
-            var namespaceDeclaration = SyntaxFactory.NamespaceDeclaration(
-                fileScopedNamespace.AttributeLists,
-                fileScopedNamespace.Modifiers,
-                fileScopedNamespace.NamespaceKeyword,
-                fileScopedNamespace.Name,
-                SyntaxFactory.Token(SyntaxKind.OpenBraceToken).WithTrailingTrivia(fileScopedNamespace.SemicolonToken.TrailingTrivia),
-                fileScopedNamespace.Externs,
-                fileScopedNamespace.Usings,
-                fileScopedNamespace.Members,
-                SyntaxFactory.Token(SyntaxKind.CloseBraceToken),
-                semicolonToken: default).WithAdditionalAnnotations(Formatter.Annotation);
-
-            // Ensure there is no errant blank line between the open curly and the first body element.
-            var firstBodyToken = namespaceDeclaration.OpenBraceToken.GetNextToken();
-            if (firstBodyToken != namespaceDeclaration.CloseBraceToken &&
-                firstBodyToken.Kind() != SyntaxKind.EndOfFileToken &&
-                HasLeadingBlankLine(firstBodyToken, out var firstBodyTokenWithoutBlankLine))
+            var nameSyntax = fileScopedNamespace.Name.WithAppendedTrailingTrivia(fileScopedNamespace.SemicolonToken.LeadingTrivia)
+                .WithAppendedTrailingTrivia(newLinePlacement.HasFlag(NewLinePlacement.BeforeOpenBraceInTypes) ? SyntaxFactory.EndOfLine(lineEnding) : SyntaxFactory.Space);
+            var openBraceToken = SyntaxFactory.Token(SyntaxKind.OpenBraceToken).WithoutLeadingTrivia().WithTrailingTrivia(fileScopedNamespace.SemicolonToken.TrailingTrivia);
+
+            if (openBraceToken.TrailingTrivia is not [.., { RawKind: (int)SyntaxKind.EndOfLineTrivia }])
```

Comment:
```suggestion
            if (openBraceToken.TrailingTrivia is not [.., SyntaxTrivia(SyntaxKind.EndOfLineTrivia)])
```
is also something you can write (i personally prefer this form).